File: IAgroplant-project/IAgroplant-back/domains/connections/application/use_cases/list_pending_connections_use_case.py
from typing import List
from domains.connections.domain.entities.connection_request import ConnectionRequest
from domains.connections.domain.repositories.connection_repository import ConnectionRepository


class ListPendingConnectionsUseCase:

    def __init__(self, repository: ConnectionRepository):
        self._repo = repository

    def execute(self, user_id: str) -> List[ConnectionRequest]:
        return self._repo.list_pending_for_user(user_id)

File: IAgroplant-project/IAgroplant-Mobile/mobile/application/use-cases/ListPendingConnectionsUseCase.ts
import { ConnectionRequest } from '../../domain/entities/connection.entity';
import { IConnectionRepository } from '../../domain/repositories/ConnectionRepository';

export class ListPendingConnectionsUseCase {
  constructor(private readonly repository: IConnectionRepository) {}

  async execute(): Promise<ConnectionRequest[]> {
    return this.repository.listPending();
  }
}
